Sources/StreamChat/Models/Attachments/ChatMessageFileAttachment.swift

```diff
@@ -33,6 +33,13 @@ public struct FileAttachmentPayload: AttachmentPayload {
             .flatMap { try? JSONEncoder.stream.encode($0) }
             .flatMap { try? JSONDecoder.stream.decode(T.self, from: $0) }
     }
+    
+    public init(title: String?, assetURL: URL, file: AttachmentFile, extraData: [String: RawJSON]?) {
```

Comment:
Opening an `init` looks safe but it actually opens the door for a miss-use 🥲 When `***AttachmentPayload` is instantiated manually there's a chance it is passed to `AnyAttachmentPayload.init` designed for  **custom non-uploadable** attachments:
```swift
init<Payload: AttachmentPayload>(payload: Payload) {
        self.init(
            type: Payload.type,
            payload: payload,
            localFileURL: nil
        )
    }
```

instead of one designed for **built-in uploadable attachments**:
```swift
init(
        localFileURL: URL,
        attachmentType: AttachmentType,
        extraData: Encodable? = nil
    ) { ... }
```

Attachment without `localURL` are not picked up for uploading and will be sent to the backend as is (with local URLs) 🥲

Reply:
IMHO that's safe, developers don't need to create built-in payload instances unless they handle the uploads themselves.
```swift
override func addAttachmentToContent(from url: URL, type: AttachmentType) throws {
        if pickingEncryptedMedia {
            pickingEncryptedMedia = false
            // Encrypt the attachment
            // Upload the attachment
            channelController?.uploadFile(type: type, localFileURL: url, completion: { result in
                do {
                    let uploadedURL = try result.get()
                    // Append already uploaded attachment
                    let attachment = AnyAttachmentPayload(
                        payload: ImageAttachmentPayload(
                            title: nil,
                            imageURL: uploadedURL,
                            extraData: nil
                        )
                    )
                    DispatchQueue.main.async {
                        self.content.attachments.append(attachment)
                    }
                } catch {
                    // Handle error
                    log.error("Failed uploading attachment: \(error)")
                }
            })
        } else {
            try super.addAttachmentToContent(from: url, type: type)
        }
    }
    ```

Reply:
They do not need but the API allows it, that's what I was saying 🙂

That's the correct way:
```swift
let attachment = try AnyAttachmentPayload(localFileURL: url, attachmentType: .image)
content.attachments.append(attachment)
```

That's the 💣 
```swift
let attachment = AnyAttachmentPayload(
    payload: ImageAttachmentPayload(
         title: nil, 
         imageURL: url, 
         extraData: nil
     )
)
content.attachments.append(attachment)
```